cmd/run.go

```diff
@@ -66,6 +73,25 @@ func (c *cmdRun) run(cmd *cobra.Command, args []string) (err error) {
 	// from sub-contexts while also attaching a reason for the abort.
 	runCtx, runAbort := execution.NewTestRunContext(lingerCtx, logger)
 
+	emitEvent := func(evt *event.Event) func() {
+		waitDone := c.gs.Events.Emit(evt)
+		return func() {
+			waitCtx, waitCancel := context.WithTimeout(globalCtx, waitEventDoneTimeout)
+			defer waitCancel()
+			if werr := waitDone(waitCtx); werr != nil {
+				logger.WithError(werr).Warn()
+			}
+		}
+	}
```

Comment:
It did previosly receive a `time.Duration`, but [we decided to replace it with a context](https://github.com/grafana/k6/pull/3112#discussion_r1218019725) because it avoided embedding one in the `System` struct, and passing one avoids using a separate timer (`time.After()`). A context is also safer since it can be based on a parent context, which ensures it can exit early. And passing both a context and a duration seems redundant.

So I see this as a helper function specific to each caller. It doesn't make sense to have it part of the `event` package, as the logic here could be different. It's only similar in both `cmd` and `js` because we essentially want to treat it in the same way. But what if the caller wants to log an `Error` instead, or fail a test, or whatever?